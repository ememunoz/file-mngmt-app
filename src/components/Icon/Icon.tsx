import React from 'react'

import svgs from './svgs'

interface Props extends React.SVGProps<SVGSVGElement> {
  name: 'main' | 'home' | 'image' | 'csv-file'
}

const Icon = ({ name, ...rest }: Props) => {
  const component = svgs[name](rest)
  return component
}

export default Icon
