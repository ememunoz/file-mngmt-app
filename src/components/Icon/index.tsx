import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

import svgs from './svgs'

interface Props extends React.SVGProps<SVGSVGElement> {
  name: 'main' | 'home' | 'image' | 'csv-file'
}

const Icon = ({ name, ...rest }: Props) => {
  const component = svgs[name](rest)
  return component
}

const styles = StyleSheet.create({})

export default Icon
