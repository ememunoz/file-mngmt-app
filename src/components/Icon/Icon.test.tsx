import { render, screen } from '@testing-library/react-native'
import React from 'react'

import Icon from './Icon'

describe('<Button />', () => {
  it('renders main correctly', () => {
    render(<Icon name="main" />)
    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('renders home correctly', () => {
    render(<Icon name="home" />)
    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('renders csv-file correctly', () => {
    render(<Icon name="csv-file" />)
    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('renders image correctly', () => {
    render(<Icon name="image" />)
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
