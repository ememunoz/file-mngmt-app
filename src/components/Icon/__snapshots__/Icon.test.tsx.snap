// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders csv-file correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="100%"
  bbWidth="100%"
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
  vbHeight={512}
  vbWidth={384}
>
  <RNSVGGroup>
    <RNSVGPath
      d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm-96 144c0 4.42-3.58 8-8 8h-8c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h8c4.42 0 8 3.58 8 8v16c0 4.42-3.58 8-8 8h-8c-26.51 0-48-21.49-48-48v-32c0-26.51 21.49-48 48-48h8c4.42 0 8 3.58 8 8v16zm44.27 104H160c-4.42 0-8-3.58-8-8v-16c0-4.42 3.58-8 8-8h12.27c5.95 0 10.41-3.5 10.41-6.62 0-1.3-.75-2.66-2.12-3.84l-21.89-18.77c-8.47-7.22-13.33-17.48-13.33-28.14 0-21.3 19.02-38.62 42.41-38.62H200c4.42 0 8 3.58 8 8v16c0 4.42-3.58 8-8 8h-12.27c-5.95 0-10.41 3.5-10.41 6.62 0 1.3.75 2.66 2.12 3.84l21.89 18.77c8.47 7.22 13.33 17.48 13.33 28.14.01 21.29-19 38.62-42.39 38.62zM256 264v20.8c0 20.27 5.7 40.17 16 56.88 10.3-16.7 16-36.61 16-56.88V264c0-4.42 3.58-8 8-8h16c4.42 0 8 3.58 8 8v20.8c0 35.48-12.88 68.89-36.28 94.09-3.02 3.25-7.27 5.11-11.72 5.11s-8.7-1.86-11.72-5.11c-23.4-25.2-36.28-58.61-36.28-94.09V264c0-4.42 3.58-8 8-8h16c4.42 0 8 3.58 8 8zm121-159L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`<Button /> renders home correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="100%"
  bbWidth="100%"
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
  vbHeight={512}
  vbWidth={576}
>
  <RNSVGGroup>
    <RNSVGPath
      d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`<Button /> renders image correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="100%"
  bbWidth="100%"
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
  vbHeight={512}
  vbWidth={448}
>
  <RNSVGGroup>
    <RNSVGPath
      d="M128 192a32 32 0 1 0-32-32 32 32 0 0 0 32 32zM416 32H32A32 32 0 0 0 0 64v384a32 32 0 0 0 32 32h384a32 32 0 0 0 32-32V64a32 32 0 0 0-32-32zm-32 320H64V96h320zM268.8 209.07a16 16 0 0 0-25.6 0l-49.32 65.75L173.31 244a16 16 0 0 0-26.62 0L96 320h256z"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`<Button /> renders main correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="100%"
  bbWidth="100%"
  focusable={false}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
  vbHeight={601.8}
  vbWidth={731.4}
>
  <RNSVGGroup>
    <RNSVGPath
      d="M389 140H136a23 23 0 0 1-23-23V23a23 23 0 0 1 23-23h253a23 23 0 0 1 23 23v94a23 23 0 0 1-23 23Z"
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M389 140H136a23 23 0 0 1-23-23V23a23 23 0 0 1 23-23h253a23 23 0 0 1 23 23v94a23 23 0 0 1-23 23ZM136 2a21 21 0 0 0-21 21v94a21 21 0 0 0 21 21h253a21 21 0 0 0 21-21V23a21 21 0 0 0-21-21Z"
      fill={4282334550}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M485 153H350a11 11 0 0 1-2-1 11 11 0 0 1-9-10V96a11 11 0 0 1 11-12h135a11 11 0 0 1 5 2 11 11 0 0 1 2 1 11 11 0 0 1 4 9v46a11 11 0 0 1-11 11Z"
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="m476 152 19-44v24a120 120 0 0 1-16 20ZM491 88l-38 64h-2q20-32 38-65a10 10 0 0 1 2 1ZM402 111a408 408 0 0 1-26 41h-14l27-33 10-8Z"
      fill={4293322470}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="m384 109-4 5-8 10-6 7a13 13 0 0 1-1-2 14 14 0 0 1 18-21 13 13 0 0 1 1 1Z"
      fill={4285293567}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="m461 119-26-10-13-5-13-4a20 20 0 0 0-11 0 27 27 0 0 0-10 6l-3 3-4 5-8 10-6 7-17 21a10 10 0 0 1-2 0l18-23 2-2 9-11 7-8v-1c5-5 12-10 19-10l13 3 13 5 26 10 7 2c1 0 1 2-1 2Z"
      fill={4293322470}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M485 153H350a11 11 0 0 1-2-1 11 11 0 0 1-9-10V96a11 11 0 0 1 11-12h135a11 11 0 0 1 5 2 11 11 0 0 1 2 1 11 11 0 0 1 4 9v46a11 11 0 0 1-11 11ZM350 86a9 9 0 0 0-9 10v46a9 9 0 0 0 7 9 9 9 0 0 0 2 0h135a9 9 0 0 0 9-9V96a9 9 0 0 0-4-8 9 9 0 0 0-1 0 9 9 0 0 0-4-2Z"
      fill={4282334550}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M217 196H23a23 23 0 0 1-23-23v-34a23 23 0 0 1 23-23h194a23 23 0 0 1 23 23v34a23 23 0 0 1-23 23Z"
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M217 196H23a23 23 0 0 1-23-23v-34a23 23 0 0 1 23-23h194a23 23 0 0 1 23 23v34a23 23 0 0 1-23 23ZM23 118a21 21 0 0 0-21 21v34a21 21 0 0 0 21 21h194a21 21 0 0 0 21-21v-34a21 21 0 0 0-21-21Z"
      fill={4282334550}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M80 177H49a6 6 0 0 1-7-6v-29a6 6 0 0 1 7-6h31a6 6 0 0 1 6 6v29a6 6 0 0 1-6 6Z"
      fill={4285293567}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M136 177h-32a6 6 0 0 1-6-6v-29a6 6 0 0 1 6-6h32a6 6 0 0 1 6 6v29a6 6 0 0 1-6 6ZM191 177h-31a6 6 0 0 1-6-6v-29a6 6 0 0 1 6-6h31a6 6 0 0 1 7 6v29a6 6 0 0 1-7 6Z"
      fill={4293322470}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M275 55h-40a4 4 0 0 1 0-8h40a4 4 0 0 1 0 8ZM361 77H234a4 4 0 0 1 0-8h127a4 4 0 0 1 0 8Z"
      fill={4291611852}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGCircle
      cx="187.2"
      cy="62.5"
      fill={4285293567}
      propList={
        Array [
          "fill",
        ]
      }
      r="26.8"
    />
    <RNSVGPath
      d="M479 313H226a23 23 0 0 1-23-23v-94a23 23 0 0 1 23-23h253a23 23 0 0 1 23 23v94a23 23 0 0 1-23 23Z"
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M479 313H226a23 23 0 0 1-23-23v-94a23 23 0 0 1 23-23h253a23 23 0 0 1 23 23v94a23 23 0 0 1-23 23ZM226 175a21 21 0 0 0-21 21v94a21 21 0 0 0 21 21h253a21 21 0 0 0 21-21v-94a21 21 0 0 0-21-21Z"
      fill={4282334550}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M325 211h-43a4 4 0 1 1 0-8h43a4 4 0 0 1 0 8Z"
      fill={4285293567}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M420 227H282a4 4 0 1 1 0-8h138a4 4 0 0 1 0 8ZM420 246H282a4 4 0 1 1 0-8h138a4 4 0 0 1 0 8ZM420 265H282a4 4 0 1 1 0-8h138a4 4 0 0 1 0 8ZM420 284H282a4 4 0 1 1 0-8h138a4 4 0 0 1 0 8Z"
      fill={4293322470}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M599 585h-14l-7-54h21v54z"
      fill={4294948534}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M599 599h-43a17 17 0 0 1 17-17l8-6 14 6h4Z"
      fill={4281282113}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="m696 559-12 8-36-40 17-11 31 43z"
      fill={4294948534}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="m705 571-35 25-1-1a17 17 0 0 1 4-23l3-9 16-4 3-2ZM654 318l1 3c8 16 10 34 7 52a3 3 0 0 0 0 2c2 2-1 11-3 17l-1 9s2 8-2 8 2 8-1 10l-10 58 39 63-14 21-58-73v-67l-9 54-1 84-25-3-10-82 9-112 14-49Z"
      fill={4281282113}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M493 299a8 8 0 0 1 11-1 7 7 0 0 1 1 1l25-9 9 11-36 9a8 8 0 0 1-9-1 7 7 0 0 1-1-10Z"
      fill={4294948534}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="m636 171-21 1-10 18-14 4-12 30 7 93s1 6-1 5-2 7-2 7l-10 38 92-8-5-41s7 0 2-5-7-3-3-9l6-10 2-29 4-41a40 40 0 0 0-15-35Z"
      fill={4285293567}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M686 375a8 8 0 0 0-2-10 7 7 0 0 0-1-1l2-27-13-6v37a8 8 0 0 0 3 9 7 7 0 0 0 10-1l1-1Z"
      fill={4294948534}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M666 201s30 72 29 99l-10 51-14-1 1-46c5-5-16-34-23-55ZM592 193s-37 69-52 89l-11 8v14l26 1s38-32 36-47 1-65 1-65Z"
      fill={4285293567}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGCircle
      cx="625.1"
      cy="140"
      fill={4294948534}
      propList={
        Array [
          "fill",
        ]
      }
      r="26.6"
    />
    <RNSVGPath
      d="m635 164-16-2c-2 0-5 0-6-3v-5a46 46 0 0 0 3-4 8 8 0 0 0-8-11l-7 2c-3 0-10-11-9-14l5-7-1-7c0-3 8 4 6 2-3-7 29-15 34-7 2 3 6 3 9 5 2 1 7 5 8 8a20 20 0 0 1 2 8c0 13-2 26-13 34"
      fill={4281282113}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="M511 594a49 49 0 0 1-39-15c-7-7-11-16-16-24l-4-8a16 16 0 0 1 2-17 15 15 0 0 1 16-5h1l6 22 3-18 1-1a78 78 0 0 1 17 11 81 81 0 0 1 30 60v1Z"
      fill={4294111986}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      d="m730 601-279 1a1 1 0 0 1 0-3h279a1 1 0 1 1 0 2Z"
      fill={4291480266}
      propList={
        Array [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
